pinot-common/src/main/java/com/linkedin/pinot/common/data/Schema.java

```diff
@@ -475,6 +522,31 @@ public SchemaBuilder addTime(@Nonnull TimeGranularitySpec incomingTimeGranularit
       return this;
     }
 
+    public SchemaBuilder addDateTime(@Nonnull String name, @Nonnull DataType dataType, @Nonnull String format,
```

Comment:
Not sure if we need so many different versions of addDateTime. Start with one, and add more as needed.

Reply:
Cleaned up addDateTime methods clutter, keeping just 1

Reply:
can we take DateTimeFieldSpec instead of individual each value. One reason why we had soo many methods for addTime was every time we enhanced TimeFieldSpec we added methods for every possible combination.